tracer/build/_build/Build.VariableGenerations.cs

```diff
@@ -91,9 +77,17 @@ void GenerateConditionVariableBasedOnGitChange(string variableName, string[] fil
                     else
                     {
                         var changedFiles = GetGitChangedFiles(baseBranch);
-
                         // Choose changedFiles that meet any of the filters => Choose changedFiles that DON'T meet any of the exclusion filters
-                        isChanged = changedFiles.Any(s => filters.Any(filter => s.StartsWith(filter, StringComparison.OrdinalIgnoreCase)) && !exclusionFilters.Any(filter => s.Contains(filter, StringComparison.OrdinalIgnoreCase)));
+
+                        foreach(var changedFile in changedFiles)
+                        {
+                            var match = codeOwners.Match("/" + changedFile);
+                            if (match?.Owners.Contains(_isChangedTeam[variableName]) == true)
+                            {
+                                Logger.Information($"File {changedFile} is owned by {_isChangedTeam[variableName]}");
+                                isChanged = true;
+                            }
+                        }
```

Comment:
nit: I think we can pull the `_isChangedTeam[variableName]` lookup out rather than doing it every time, and we can bail as soon as we find a match
```suggestion
                        var teamName = _isChangedTeam[variableName];
                        foreach(var changedFile in changedFiles)
                        {
                            var match = codeOwners.Match("/" + changedFile);
                            if (match?.Owners.Contains(teamName) == true)
                            {
                                Logger.Information($"File {changedFile} is owned by {teamName}");
                                isChanged = true;
                                break;
                            }
                        }
```

Reply:
I had the break statement previously and removed it in order to get all the log messages related to the changed files. I could add the break statement though to improve performance even though we would not get the full list of changed files. WDYT?
I will add teamName. Thanks!

Reply:
> I had the break statement previously and removed it in order to get all the log messages related to the changed files

I see 🤔 I think in practice, it's unlikely we're really going to ever look at those logs, and this stage is a blocker for subsequent stages, so it's _probably_ preferable to be slightly faster? 🤔

Reply:
Yes, that makes sense, thanks!